docs/astro/src/content/docs/guide/language/concepts/slint-language.mdx

```diff
@@ -4,4 +4,150 @@ title: Slint Language
+of text, images, colors, animations, and so on. Even though it's a mirage, the buttons and elements do not
+really exist in the physical world, it's meant to look and behave as if it does. Buttons have pressed effects,
+lists can be flicked and behave as if they had real inertia. When being designed they are described in terms
+of UI components, forms and views. 
+
+Meanwhile the world of code is a quite different abstraction. It's made up of functions, variables, and so on.
+Even when some aspects of a UI exist such as buttons and menus they also go hand in hand with a lot of code to
+manage the implementation details.
+
+```js
+const button = document.createElement('button');
+button.textContent = 'Click me';
+document.body.appendChild(button);
+```
+
+Take this simple web example. A button is created. Then a property to show 'Click me' text is text. At this point
+technically the button exists, but it won't show up as its not attached to anything. So the final line
+makes it a parent of the main view. 
+
+```js
+const buttonWithListener = document.createElement('button');
+buttonWithListener.textContent = 'Click me';
+buttonWithListener.addEventListener('click', () => {
+    console.log('Button clicked!');
+});
+document.body.appendChild(buttonWithListener);
+```
+
+In this second example a button is created and an event listener is added. Within that is a callback
+function to log out that the button is pressed. It's a lot of code to so simple things.
+
+Its quite abstract. For example once a few more buttons and components are added its almost impossible
+to be able to think how the real interface would look. It's hard to think about the design of a UI
+using this kind of code.
+
+It's also too complex to edit without understanding how to code. This means UI designers cannot work
+hands on to ensure all their design intent is implemented. They are forced to use other tools and frameworks
+to crate prototypes and design guides that may look or behave differently to the software used to implementat
+the production UI. It should not have to be this way.
+
+## Declartive style
+
+There have been attempts to make describing a UI in code more declarative. For example [React](https://reactjs.org/)
+and [SwiftUI](https://developer.apple.com/xcode/swiftui/).
+
+
+```jsx
+function ContentView() {
+  return (
+    <p style={{ fontSize: '2rem', color: 'green' }}>
+      Hello World
+    </p>
+  );
+}
+```
+
+```swift
+struct ContentView: View {
+    var body: some View {
+        Text("Hello World")
+            .font(.title)
+            .foregroundColor(.green)
+    }
+}
+```
+
+These languages take normal code and let it be used in a declartive way. But it's 
+still functions with arguments that act as properties. It is simpler and behaviour
+such as parent child relationships can be inferred. 
+
+For Slint instead of tweaking a normal language to be more declarative we instead 
+created a pure declarative language from the ground up.
```

Comment:
```suggestion
For Slint, instead of tweaking a normal language to be more declarative we 
created a pure declarative language from the ground up.
```